openingd/openingd.c

```diff
@@ -822,6 +822,11 @@ static u8 *funder_channel(struct state *state,
 					   state->feerate_per_kw,
 					   state->localconf.channel_reserve);
 
+fail_2:
```

Comment:
Yes, the `if`s are not necessary within `fail_2`.

- Either you initialize with 0/NULL and can do an `if(ptr)`  within just one `fail` handler, because you wont get valgrind `jump or move depends on uninitialized value` errors for the `if(ptr)` access. 
- Or you have two `fail_2` gotos, that 'know' at which point the pointers have to be initialized. In this case we can skip the `if(ptr)`s before the `free(ptr)`.

I will push the next update to remove the unnecessary `if(ptr)` checks in fail_2.
If you prefer just one `fail` handler, we need to initialize with NULL for valgrind mem access check.